tests/functional/configured_endpoint_urls/test_configured_endpoint_url.py

```diff
@@ -0,0 +1,146 @@
+        else:
+            section_str += f"{key}={value}\n"
+    return section_str + "\n"
+
+
+def create_cases():
+    with open(ENDPOINT_TESTDATA_FILE) as f:
+        test_suite = json.load(f)['testSuites'][0]
+
+    for test_case_data in test_suite['endpointUrlTests']:
+        yield pytest.param(
+            {
+                'service': test_case_data['service'],
+                'profile': test_case_data['profile'],
+                'expected_endpoint_url': test_case_data['output'][
+                    'endpointUrl'
+                ],
+                'client_args': test_suite['client_configs'].get(
+                    test_case_data['client_config'], {}
+                ),
+                'config_file_contents': get_config_file_contents(
+                    test_case_data['profile'], test_suite
+                ),
+                'environment': test_suite['environments'].get(
+                    test_case_data['environment'], {}
+                ),
+            },
+            id=test_case_data['name']
+        )
+
+
+def get_config_file_contents(profile_name, test_suite):
+    profile = test_suite['profiles'][profile_name]
+
+    profile_str = dict_to_ini_section(
+        profile,
+        section_header=f"profile {profile_name}",
+    )
+
+    services_section_name = profile.get('services', None)
+
+    if services_section_name is None:
+        return profile_str
+
+    services_section = test_suite['services'][services_section_name]
+
+    service_section_str = dict_to_ini_section(
+        services_section,
+        section_header=f'services {services_section_name}',
+    )
+
+    return profile_str + service_section_str
+
+
+def _normalize_endpoint(url):
+    split_endpoint = urlparse.urlsplit(url)
+    actual_endpoint = f"{split_endpoint.scheme}://{split_endpoint.netloc}"
+    return actual_endpoint
+
+
+SERVICE_TO_OPERATION = {'s3api': 'list-buckets', 'dynamodb': 'list-tables'}
+
+
+class TestConfiguredEndpointUrl:
+    def assert_endpoint_used(
+        self, cli_runner_result, test_case
+    ):
+
+        aws_request = cli_runner_result.aws_requests[0]
+        assert test_case['expected_endpoint_url'] == _normalize_endpoint(aws_request.http_requests[0].url), test_case
+
+    def _create_command(self, test_case):
+        service = test_case['service']
+        if test_case['service'] == 's3':
+            service = 's3api'
+
+        cmd = [
+            service,
+            SERVICE_TO_OPERATION[service],
+            '--profile',
+            f'{test_case["profile"]}'
+        ]
+        if test_case['client_args']:
+            cmd.extend([
+                    '--endpoint-url',
+                    f'{test_case["client_args"]["endpoint_url"]}'
+                ]
+            )
+
+        return cmd
+
+    @pytest.mark.parametrize('test_case', create_cases())
+    def test_resolve_configured_endpoint_url(self, test_case):
+        session_stubber = SessionStubber()
```

Comment:
I think I can move the creation and config of the runner into a helper fixture, will work on that. Wanted to be sure this was functional!